api/types/events/resource_ids.go

```diff
@@ -32,3 +36,73 @@ func ResourceIDs(resourceIDs []types.ResourceID) []ResourceID {
 	}
 	return out
 }
+
+// ToEventResourceAccessIDs converts []types.ResourceAccessID to []ResourceAccessID for events.
+func ToEventResourceAccessIDs(in []types.ResourceAccessID) []ResourceAccessID {
+	out := make([]ResourceAccessID, 0, len(in))
+	for _, r := range in {
+		out = append(out, ToEventResourceAccessID(r))
+	}
+	return out
+}
+
+// ToEventResourceAccessID converts a types.ResourceAccessID to a ResourceAccessID for events.
+func ToEventResourceAccessID(in types.ResourceAccessID) ResourceAccessID {
+	out := ResourceAccessID{
+		Id: ResourceID{
+			ClusterName:     in.Id.ClusterName,
+			Kind:            in.Id.Kind,
+			Name:            in.Id.Name,
+			SubResourceName: in.Id.SubResourceName,
+		},
+	}
+
+	c := in.Constraints
+	if c == nil {
+		return out
+	}
+
+	details := c.GetDetails()
+	if details == nil {
+		return out
+	}
+
+	switch details.(type) {
+	// AWS Console constraints variant
+	case *types.ResourceConstraints_AwsConsole:
+		aws := c.GetAwsConsole()
```

Comment:
If you use `switch d := details.(type)`, `d` will have a different type in each `case`, eliminating the need to use `GetAwsConsole` (which internally once more performs the typecasting logic) and allowing you to operate directly with a safely typed object.